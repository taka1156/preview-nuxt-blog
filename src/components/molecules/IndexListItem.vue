<template>
  <div>
    <a class="index-list-item" :href="`#${t.anchor}`" v-scroll-to="`#${t.anchor}`">
      {{t.index}}. {{ t.escapedText }}
    </a>
  </div>
</template>

<script>
export default {
  name: 'IndexListItem',
  props: {
    t: {
      type: Object,
      default: () => {},
      required: true
    }
  }
};
</script>

<style scoped>
.index-list-item {
  color: rgba(40, 167, 69, 0.9);
  font-size: 13px;
}

@media (min-width: 790px) {
  .index-list-item {
    font-size: 20px;
  }
}
</style>
