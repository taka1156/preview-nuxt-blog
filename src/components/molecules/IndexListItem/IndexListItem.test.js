import { shallowMount } from '@vue/test-utils';
import IndexListItem from './IndexListItem.vue';
import { dummyTocs } from '@/__testdata__/testdata.js';

describe('IndexList', () => {
  const indexListItem = propsData =>
    shallowMount(IndexListItem, {
      propsData: {
        ...propsData
      }
    });

  it('IndexListItem初期値: t', () => {
    const wrapper = indexListItem({ t: dummyTocs[0] });
    // t
    expect(wrapper.vm.$options.props.t.required).toBe(true);
    expect(wrapper.vm.t).toBe(dummyTocs[0]);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
