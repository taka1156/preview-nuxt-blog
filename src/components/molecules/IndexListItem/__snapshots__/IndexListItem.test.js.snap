// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexList IndexListItem初期値: t 1`] = `
"<div><a href=\\"#anchor_1\\" class=\\"index-list-item\\">
    1.この文章はダミーです。
  </a></div>"
`;
